src/MassTransit/Logging/Diagnostics/LogContextActivityExtensions.cs

```diff
@@ -82,6 +82,19 @@ public static class LogContextActivityExtensions
                         || context.TransportHeaders.TryGetHeader(MessageHeaders.MessageId, out messageIdHeader))
                     && messageIdHeader is string text)
                     activity.SetTag(DiagnosticHeaders.Messaging.TransportMessageId, text);
+
+                if (context.GetConversationId().HasValue)
+                    activity.SetTag(DiagnosticHeaders.Messaging.ConversationId, context.GetConversationId()!.Value.ToString("D"));
+                if (context.GetCorrelationId().HasValue)
+                    activity.SetTag(DiagnosticHeaders.CorrelationId, context.GetCorrelationId()!.Value.ToString("D"));
+                if (context.GetRequestId().HasValue)
+                    activity.SetTag(DiagnosticHeaders.RequestId, context.GetRequestId()!.Value.ToString("D"));
+                if (context.GetInitiatorId().HasValue)
```

Comment:
Why calling a method here? Makes no sense to me.

Reply:
Honestly this is a bad idea, this shouldn't be done here.

Reply:
Yeah, no I get what you're trying to do, but no.

Reply:
So there is no way to get these data during the receive phase?

Reply:
There is not.

Reply:
Thank you for your feedback. I reverted this part.